src/item.cpp

```diff
@@ -4221,60 +4221,62 @@ const item * item::gunmod_find( const itype_id& mod ) const
     return const_cast<item *>( this )->gunmod_find( mod );
 }
 
-bool item::gunmod_compatible( const item& mod, bool alert, bool effects ) const
```

Comment:
None. When one installs a gunmod in ...

the current master:

- `gunmod_compatible( effects = false )` gets called for each gun to filter compatible ones
- `gunmod_compatible( effects = true )` gets called for selected gun to reject it if it's loaded (fiasco).

this PR:

- `gunmod_compatible()` gets called for each gun only once and filters out incompatible or loaded ones. It shows an appropriate denial message in either case.